src/main/java/com/microsoft/azure/AzureVMCloud.java

```diff
@@ -476,8 +476,9 @@ public Node call() throws Exception {
                                             AzureVMManagementServiceDelegate.setVirtualMachineDetails(
                                                     agentNode, template);
                                             Jenkins.getInstance().addNode(agentNode);
-                                            if (agentNode.getAgentLaunchMethod().equalsIgnoreCase("SSH")) {
-                                                agentNode.toComputer().connect(false).get();
+                                            Computer computer = agentNode.toComputer();
+                                            if (agentNode.getAgentLaunchMethod().equalsIgnoreCase("SSH") && computer != null) {
```

Comment:
Assuming the `azureComputer` is the right variable, this `computer != null` check would be redundant AFAICT